From ac1bd4fc3bfd75a23612acaee36a0e6e5d6bc2a9 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Mon, 8 Aug 2022 10:04:31 +0800
Subject: [PATCH] sw64: switch to inline _copy_{to,from}_user()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: ac1bd4fc3bfd75a23612acaee36a0e6e5d6bc2a9
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5PN9S

--------------------------------

Switch to inline version of _copy_{to,from}_user() to improve
performance.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/uaccess.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/sw_64/include/asm/uaccess.h b/arch/sw_64/include/asm/uaccess.h
index ceacfaa07cfb..730121aad184 100644
--- a/arch/sw_64/include/asm/uaccess.h
+++ b/arch/sw_64/include/asm/uaccess.h
@@ -292,6 +292,8 @@ raw_copy_to_user(void __user *to, const void *from, unsigned long len)
 {
 	return __copy_user((__force void *)to, from, len);
 }
+#define INLINE_COPY_FROM_USER
+#define INLINE_COPY_TO_USER
 
 extern long __clear_user(void __user *to, long len);
 
-- 
2.33.0

